gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/transformation/functions/TransformGraphTransaction.java

```diff
@@ -30,36 +33,43 @@
 
 /**
  * Transformation map function for graph transactions.
+ *
+ * Since {@link GraphTransaction} is only implemented for EPGM, this can only be used with EPGM graphs.
+ *
+ * @param <G> type of the graph head
+ * @param <V> the vertex type
+ * @param <E> the edge type
  */
-public class TransformGraphTransaction implements MapFunction<GraphTransaction, GraphTransaction> {
+public class TransformGraphTransaction<G extends GraphHead, V extends Vertex, E extends Edge>
```

Comment:
Why was this generified if it is only supported by EPGM-based instances?

Reply:
Because checking the generic (`a instanceof A<C>`) is not possible, but needed in `ApplyTransformation`. Instead i cast the instance of the generic here.